File: CrossDomainPGELLAv1.0/DlogPiDThetaNAC.m
function [der1,der2] = DlogPiDThetaNAC(policy, x,u,param)

% Programmed by Jan Peters (jrpeters@usc.edu).
N = param.param.N;
M = param.param.M;

sigma = max(policy.sigma,0.00001);
k = policy.theta;
xx = x;
der1 = [];
der2 = [];
for i=1:M
    der1 = [der1;(u(i)-k(1+N*(i-1):N*(i-1)+N)'*xx)*xx/(sigma(i)^2)];
%     der2 = [der2;((u(i)-k(1+N*(i-1):N*(i-1)+N)'*xx)^2-sigma(i)^2)/(sigma(i)^3)];
end

% der = [der1;der2]

File: src/vertical_control/scripts/pg_ella/DlogPiDThetaNAC.py
# -*- coding: utf-8 -*-
"""
Created on Tue Aug 25 16:20:41 2015

@author: brownyoda
"""

import numpy as np


def DlogPiDThetaNAC(policy, x, u, param):

    # Programmed by Jan Peters
    N = param.param.N
    M = param.param.M

    sigma = np.max(policy.sigma, 0.00001)
    k = policy.theta
    der = np.empty(shape=(0, N))
    for i in range(M):
        der = np.concatenate((der, np.dot((u[i]-np.dot(k[N*(i):N*(i)+N].conj().T, x)), x.conj().T) / (sigma[i]**2)))

    return der